src/utils/workflow-utils.ts

```diff
@@ -156,3 +157,24 @@ export const modifyExistingCaptureDraftSpec = async (
         spec_type: 'capture',
     });
 };
+
+// Common materialization field selection checks
+export const evaluateRequiredIncludedFields = (
+    constraintType: ConstraintTypes
+): boolean => {
+    return (
+        constraintType === ConstraintTypes.FIELD_REQUIRED ||
+        constraintType === ConstraintTypes.LOCATION_REQUIRED
+    );
+};
+
+export const evaluateRecommendedIncludedFields = (
+    constraintType: ConstraintTypes
+): boolean => {
+    return (
+        constraintType === ConstraintTypes.FIELD_REQUIRED ||
+        constraintType === ConstraintTypes.LOCATION_REQUIRED ||
+        constraintType === ConstraintTypes.LOCATION_RECOMMENDED ||
+        constraintType === ConstraintTypes.FIELD_OPTIONAL
+    );
+};
```

Comment:
If the `contraintType` checks are pretty locked down then I think we should was have `evaluateRecommendedIncludedFields` call `evaluateRequiredIncludedFields` to cover the field and location checks. That way they are always in sync.